chef/automate
# SSH User Check API

### :nut_and_bolt: Description: What code changed, and why?
 Added the SSH User Check API for the Bastion as if we hit this api with the required body then it will check If the user entered has the SSH access to the node as well as if password is given does it has the access to run the sudo cammand on that node.


### :chains: Related Resources

### :+1: Definition of Done

### :athletic_shoe: How to Build and Test the Change
 - Take the pull of this PR
 - rebuild components/automate-cli/
 - Add the this rebuild to your hab package 
 - Install that cli build on the ubuntu or any linux  supported by Automate
 - Run `chef-automate verify serve`
 - The Required parameter or hitting the request are 
 - ```
    "ip": "ENTER VALID IP ",
    "user_name": "ENTER THE VALID USER NAME",
    "ssh_port": "ENTER THE PORT FOR SSH CONNECTION",
    "private_key": "--- ENTER VALID PRIVATE KEY ----",
    "sudo_password":"ENTER SUDO PASSWORD FOR THE USER"
### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
- Success Response
  https://progresssoftware.sharepoint.com/:i:/s/ChefCoreC/EZzC6bAXfGFFlA0D63iCK8sBenBPvp7ZB7Kssvooim69Iw?e=rLfKhK


- If the User Doesn't have the SSH and Password privileges
  https://progresssoftware.sharepoint.com/:i:/s/ChefCoreC/Eb38ar4nFadLhi-EQvMqh4ABEWzlQU1-gflLW3mDJ-rdzw?e=x3lUud

- Demo Video for the SSH User Check API
   https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EfQtzyvStuRHvpT9Medjb7cB6b-i2NKFL7pbLVtxqbTSvQ?e=uEUXhz
 
- SSH success from one node to another 
    https://progresssoftware.sharepoint.com/:i:/s/ChefCoreC/EY9A2acc2x5NmQTzL1ScducBhIRShVhGj-hEodrXfU1Qvg?e=mtGlw2
